server/battles/BattleFlowProcessor.cpp

```diff
@@ -342,6 +342,21 @@ void BattleFlowProcessor::activateNextStack(const CBattleInfoCallback & battle)
 
 bool BattleFlowProcessor::tryMakeAutomaticAction(const CBattleInfoCallback & battle, const CStack * next)
 {
+	bool actionPerformed = tryActivateMoralePenalty(battle, next) || tryActivateBerserkPenalty(battle, next) || tryAutomaticActionOfWarMachines(battle, next);
+
+	if (!actionPerformed) {
+		stackTurnTrigger(battle, next); //various effects
+
+		if(next->fear)
+			{
+				makeStackDoNothing(battle, next); //end immediately if stack was affected by fear
+				return true;
+			}
+		}
+	return actionPerformed;
+}
```

Comment:
Formatting is broken.
I've added clang-format configuration recently, you can try using it if your IDE supports it.

Also - perhaps simplify this method like this? Right now it is a bit hard to follow the flow
```suggestion
	if (tryActivateMoralePenalty(battle, next))
		return true;

	if (tryActivateBerserkPenalty(battle, next))
		return true;
	
	if (tryAutomaticActionOfWarMachines(battle, next))
		return true;
		
	stackTurnTrigger(battle, next); //various effects

	if(next->fear)
	{
		makeStackDoNothing(battle, next); //end immediately if stack was affected by fear
		return true;
	}

	return false;
}
```

Reply:
Fixed